src/skmultiflow/meta/learn_nse.py

```diff
@@ -0,0 +1,286 @@
+import copy as cp
+import numpy as np
+from skmultiflow.core.base import StreamModel
+from sklearn.tree import DecisionTreeClassifier
+
+
+class LearnNSE(StreamModel):
+    """
+    Ensemble of classifiers-based approach for incremental learning of concept
+    drift, characterized by nonstationary environments (NSEs), where the
+    underlying data distributions change over time. It learns from consecutive
+    batches of data that experience constant or variable rate of drift,
+    addition or deletion of concept classes, as well as cyclical drift.</p>
+
+    References
+    ----------
+    Ryan Elwell and Robi Polikar. Incremental learning of concept drift in
+    non-stationary environments. IEEE Transactions on Neural Networks,
+    22(10):1517-1531, October 2011. ISSN 1045-9227. URL
+    http://dx.doi.org/10.1109/TNN.2011.2160459
+
+    Parameters
+    ----------
+    base_estimator: StreamModel or sklearn.BaseEstimator (default=DecisionTreeClassifier)
+        This is the base estimator, each member of the ensemble is an instance of the base estimator.
+    ensemble_size: int (default=15)
+        The number of ensembles to keep.
+    period: int (default=250)
+        The size of the training window (batch), in other words, how many instances are kept for training.
+    crossing_point: float (default=0.5)
+        Halfway crossing point of the sigmoid function controlling the number of previous
+        periods taken into account during weighting.
+    slope: float (default=0.5)
+        Slope of the sigmoid function controlling the number
+        of previous periods taken into account during weighting.
+    pruning: string (default=None)
+        Classifiers pruning strategy to be used.
+        pruning=None: Don't prune classifiers
+        pruning='age': Age-based
+        pruning='error': Error-based
+    """
+
+    def __init__(self,
+                 base_estimator=DecisionTreeClassifier(),
+                 period=250,
+                 slope=0.5,
+                 crossing_point=10,
+                 ensemble_size=15,
+                 pruning=None):
+        self.ensemble = []
+        self.ensemble_weights = []
+        self.bkts = []
+        self.wkts = []
+        self.buffer = []
+        self.period = period
+        self.slope = slope
+        self.crossing_point = crossing_point
+        self.ensemble_size = ensemble_size
+        self.pruning = pruning
+        self.X_batch = []
+        self.y_batch = []
+        self.instance_weights = []
+        self.base_estimator = cp.deepcopy(base_estimator)
+        self.classes = None
+
+    def fit(self, X, y, classes=None, weight=None):
+        raise NotImplementedError
+
+    def __fit(self, estimator, X, y, sample_weight=None, classes=None):
+        try:
```

Comment:
The `partial_fit` method is the default in `scikit_multiflow`, my suggestions are:

```python
    def __fit(self, estimator, X, y, classes=None, sample_weight=None):
        try:
            estimator.partial_fit(X, y, weight=sample_weight, classes=classes)
        except (NotImplementedError, TypeError):
            estimator.fit(X, y, weight=sample_weight, classes=classes)
```

Reply:
I don't know how to use weight in scikit multiflow. If I pass 1/m. It returns all 0. as I write when I use Hoeffding Tree

Reply:
I reviewed the implementation of HoeffdingTrees and NaiveBayes and found no errors. What you describe is related to the fact that since we are normalizing the `weights` it means that the contribution of each sample is very small and the model is updated in small increments.

From your example above you can test this by changing:
```python
weight = np.ones(m) / (m//10)
```

Notice that there was indeed a bug in HoeffdingTree when passing more than 1 sample, but the fix is in place in the master. Please bring those changes to your branch.